contracts/contracts/harvest/BaseHarvester.sol

```diff
@@ -328,31 +428,255 @@ abstract contract BaseHarvester is Governable {
     }
 
     /**
-     * @dev Swap all supported swap tokens for stablecoins via Uniswap. And send the incentive part
-     * of the rewards to _rewardTo address.
-     * @param _rewardTo Address where to send a share of harvest rewards to as an incentive
-     *      for executing this function
+     * @dev Swap a reward token for the base token on the configured
+     *      swap platform. The token must have a registered price feed
+     *      with the price provider
+     * @param _swapToken Address of the token to swap
+     * @param _rewardTo Address where to send the share of harvest rewards to
      */
-    function _swap(address _rewardTo) internal {
-        address[] memory allStrategies = IVault(vaultAddress)
-            .getAllStrategies();
-
-        for (uint256 i = 0; i < allStrategies.length; i++) {
-            IStrategy strategy = IStrategy(allStrategies[i]);
-            address[] memory rewardTokenAddresses = strategy
-                .getRewardTokenAddresses();
-
-            for (uint256 j = 0; j < rewardTokenAddresses.length; j++) {
-                _swap(rewardTokenAddresses[j], _rewardTo);
-            }
+    function _swap(address _swapToken, address _rewardTo) internal virtual {
+        RewardTokenConfig memory tokenConfig = rewardTokenConfigs[_swapToken];
+
+        /* This will trigger a return when reward token configuration has not yet been set
+         * or we have temporarily disabled swapping of specific reward token via setting
+         * doSwapRewardToken to false.
+         */
+        if (!tokenConfig.doSwapRewardToken) {
+            return;
+        }
+
+        address priceProvider = IVault(vaultAddress).priceProvider();
+        uint256 balance = IERC20(_swapToken).balanceOf(address(this));
+
+        if (balance == 0) {
+            return;
+        }
+
+        if (tokenConfig.liquidationLimit > 0) {
+            balance = Math.min(balance, tokenConfig.liquidationLimit);
+        }
+
+        // This'll revert if there is no price feed
+        uint256 oraclePrice = IOracle(priceProvider).price(_swapToken);
+
+        // Oracle price is 1e18
+        uint256 minExpected = (balance *
+            (1e4 - tokenConfig.allowedSlippageBps) * // max allowed slippage
+            oraclePrice).scaleBy(
+                baseTokenDecimals,
+                Helpers.getDecimals(_swapToken)
+            ) /
+            1e4 / // fix the max slippage decimal position
```

Comment:
You can use `StableMath`'s `mulTruncate` && `mulTruncateScale` that will make this operation a bit more readable:
```
balance
     .mulTruncateScale(1e4 - tokenConfig.allowedSlippageBps, 1e4)
     .mulTruncate(oraclePrice)
    .scaleBy(
        baseTokenDecimals,     
        Helpers.getDecimals(_swapToken)
    )
```

Reply:
I'm not sure if chaining these calls will work as expected due to rounding. But I can probably nest them

Reply:
Okay, we do the `balance * slippage * price` multiplication first and then scale it by base token's decimals and then adjust the decimals. 

Changing the order gives us a less accurate value due to rounding. But let me test this a bit more. For now, I've not committed this change

Reply:
So, one potential bug with your suggested approach is that it won't work for tokens with decimals of <18 and a really low price. Let's say `NotARugToken` uses 6 decimals, if the Oracle price is less than 1e6, it's gonna be rounded down to 0. But this is probably an edge case.

Right now none of our reward tokens use <18 decimals, but I don't wanna make that assumption for future as well. And I guess I'd prefer a bit higher accuracy than readability for this line. But let me know if you have any other suggestions on this one